File: solutions/cpp/vehicle-purchase/1/vehicle_purchase.cpp
#include "vehicle_purchase.h"

namespace vehicle_purchase {

// needs_license determines whether a license is needed to drive a type of
// vehicle. Only "car" and "truck" require a license.
bool needs_license(std::string kind) {
    return kind == "car" || kind == "truck";
}

// choose_vehicle recommends a vehicle for selection. It always recommends the
// vehicle that comes first in lexicographical order.
std::string choose_vehicle(std::string option1, std::string option2) {
    std::string chosen_vehicle =  option1 > option2 ? option2 : option1;
    return chosen_vehicle + " is clearly the better choice." ;
}

// calculate_resell_price calculates how much a vehicle can resell for at a
// certain age.
double calculate_resell_price(double original_price, double age) {
    if(age < 3){
        return original_price * 0.8;
    }else if(age >= 10){
        return original_price * 0.5;
    }else{
        return original_price * 0.7;
    }
}

}  // namespace vehicle_purchase

File: solutions/javascript/vehicle-purchase/1/vehicle-purchase.js
//@ts-check

/**
 * Determines whether or not you need a licence to operate a certain kind of vehicle.
 *
 * @param {string} kind
 * @returns {boolean} whether a license is required
 */
export function needsLicense(kind) {
  return (kind == 'car') || (kind == 'truck');
}

/**
 * Helps choosing between two options by recommending the one that
 * comes first in dictionary order.
 *
 * @param {string} option1
 * @param {string} option2
 * @returns {string} a sentence of advice which option to choose
 */
export function chooseVehicle(option1, option2) {
  var sentence = ' is clearly the better choice.';
  return (option1 < option2 ? option1 + sentence : option2 + sentence);
}

/**
 * Calculates an estimate for the price of a used vehicle in the dealership
 * based on the original price and the age of the vehicle.
 *
 * @param {number} originalPrice
 * @param {number} age
 * @returns expected resell price in the dealership
 */
export function calculateResellPrice(originalPrice, age) {
  if(age < 3){
   return originalPrice * 0.8; 
  }else if(age > 10){
    return originalPrice * 0.5; 
  }else{
    return originalPrice * 0.7; 
  }
}
